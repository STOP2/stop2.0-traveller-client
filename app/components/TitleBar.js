import React, { Component } from 'react'
import { View } from 'react-native'
import { DefaultText, BoldText } from '../components/textComponents'

import styles from '../styles/stylesheet'

export class TitleBar extends Component {
    render()
    {
        return (<DefaultText style={styles.title}>{this.props.title}</DefaultText>)
    }
}

TitleBar.propTypes = {title: React.PropTypes.string.isRequired}

export class BoldTitleBar extends Component {
    render()
    {
        return (<View style={this.props.noBorder ? styles.mainTitleCont : [styles.mainTitleCont, styles.bottomBorder]}>
                  <BoldText style={styles.mainTitle}>{this.props.title}</BoldText>
                </View>)
    }
}

BoldTitleBar.defaultProps = { noBorder: false }

BoldTitleBar.propTypes = {
    title: React.PropTypes.string.isRequired,
    noBorder: React.PropTypes.bool
}
