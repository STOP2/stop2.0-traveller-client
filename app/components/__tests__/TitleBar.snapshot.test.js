import 'react-native'
import React from 'react'
import { TitleBar } from '../TitleBar'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('TitleBar component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<TitleBar title="otsikko" />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
