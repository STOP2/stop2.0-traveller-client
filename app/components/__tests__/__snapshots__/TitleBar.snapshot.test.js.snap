exports[`BoldTitleBar component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#007AC9",
        "height": 50,
        "justifyContent": "center",
      },
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 0.5,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-medium",
        },
        Object {
          "color": "white",
          "fontSize": 25,
        },
      ]
    }>
    otsikko
  </Text>
</View>
`;

exports[`BoldTitleBar component renders correctly without border 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#007AC9",
      "height": 50,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "gotham-rounded-medium",
        },
        Object {
          "color": "white",
          "fontSize": 25,
        },
      ]
    }>
    otsikko
  </Text>
</View>
`;

exports[`TitleBar component renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "gotham-rounded-book",
      },
      Object {
        "backgroundColor": "#007AC9",
        "color": "white",
        "fontSize": 25,
        "padding": 15,
      },
    ]
  }>
  otsikko
</Text>
`;
